proposals/nnnn-sendable-inference.md

```diff
@@ -0,0 +1,274 @@
+
+Also note that, in some cases, the type wll not be `Sendable` because a concrete type---not a type parameter---lacks a suitable `Sendable` conformance. For example:
+
+```swift
+class Ref<T> { } // not Sendable
+
+struct Generic2<T> { // no Sendable conformance at all because Ref<T> is always non-Sendable
+  var ref: Ref<T> 
+}
+```
+
+In such cases, no `Sendable` conformance will be inferred.
+
+### Compiler support to help find all `Sendable` types
+
+As existing Swift code embraces `Sendable`, it can be a chore to audit each and every `public` type to determine whether it should be `Sendable`. A missing `Sendable` conformance in a library makes it harder for clients to use Swift's concurrency model with that library. We propose the addition of a new compiler flag, `-require-explicit-sendable`, that requires each `public` (or `open`) non-protocol type to either be `Sendable` or be marked explicitly non-`Sendable` with the syntax proposed above. The compiler will produce a warning for each type that does not meet these requirements. As an implementation matter, the compiler should provide Fix-Its to help with these conformances. For example, if the `Generic` type from the previous section were `public` and had no `Sendable` conformance , the compiler could provide Fix-Its options to either make it conditionally `Sendable` or to make it explicitly non-`Sendable`.
+
+```swift
+public struct Generic<T: P> { // warning: public type `Generic` is neither `Sendable` nor explicitly non-`Sendable`
+    // note: add conditional conformance to `Sendable` if this values of this type can be copied across concurrent tasks
+    // note: add explicit non-Sendable annotation to suppress this warning
+  var value: T? = nil
+  var assocValues: [T.A] = []
+}
+```
+
+With such a flag, a user can conduct an audit of all of the public types for `Sendable`, using Fix-Its to make most of the required source code changes. The flag could either be removed (possibly along with the explicit non-`Sendable` annotations) or left in place to ensure that `Sendable` is considered for future public types.
+
+## Source compatibility
+
+This proposals alters `Sendable` conformances in several ways that can have impact on source compatibility.This proposa is a mix of additions and removals. The additions (e.g., conditional `Sendable` conformances for non-`public` types) aren't likely to break much code, because adding protocol conformances rarely does, and the new compiler flag is opt-in.
```

Comment:
```suggestion
This proposal alters `Sendable` conformances in several ways that can have an impact on source compatibility. This proposal is a mix of additions and removals. The additions (e.g., conditional `Sendable` conformances for non-`public` types) aren't likely to break much code, because adding protocol conformances rarely does, and the new compiler flag is opt-in.
```